packages/react-composites/src/composites/CallComposite/components/MediaGallery.tsx

```diff
@@ -60,11 +65,58 @@ export const MediaGallery = (props: MediaGalleryProps): JSX.Element => {
     };
   }, [cameraSwitcherCallback, cameraSwitcherCameras]);
 
+  const [augmentedParticipants, setAugmentedParticipants] = useState<VideoGalleryRemoteParticipant[] | undefined>(
+    undefined
+  );
+
+  // Async function to retrieve custom data model information to replace display name in remote tiles with camera on.
+  const fetchAvatarPersonaDataAsync = memoizeFunction(
+    async (
+      onFetchAvatarPersonaData: AvatarPersonaDataCallback,
+      participants: VideoGalleryRemoteParticipant[]
+    ): Promise<VideoGalleryRemoteParticipant[] | undefined> => {
+      if (props.onFetchAvatarPersonaData) {
+        await Promise.all(
+          participants.map(async (participant) => {
+            const newParticipantData = await onFetchAvatarPersonaData(participant.userId);
+            participant.displayName = newParticipantData.text ? newParticipantData.text : participant.displayName;
+          })
+        );
+        return participants;
+      }
+      return;
+    }
+  );
+
+  // if we have the onFetchAvatarPersonaData callback set go through and edit the remote participant data.
+  useEffect(() => {
+    // flag to stop race conditions caused by participants joining the call
+    let newestFetch = true;
+    const augmentDisplayName = async (): Promise<void> => {
+      if (props.onFetchAvatarPersonaData) {
+        const tempParticipants = await fetchAvatarPersonaDataAsync(
+          props.onFetchAvatarPersonaData,
+          videoGalleryProps.remoteParticipants as VideoGalleryRemoteParticipant[]
```

Comment:
No need to typecast here. 
`videoGalleryProps.remoteParticipants` should be of time `VideoGalleryRemoteParticipant[]`

Reply:
If you don't type cast here it thinks its type `never` for some reason